R/pairwise_cd_equal.R

```diff
@@ -0,0 +1,77 @@
+#' Approximate a pairwise Cramér distance between
+#' two equally-spaced vectors of quantiles
+#'
+#' @param q_F vector containing the quantiles of F
+#' @param q_G vector containing the quantiles of G
+#' @param approx_rule string specifying which formula to use
+#' for approximation. Valid rules are "first_rule" and "second_rule".
+#' See Details for more information.
+#' @return a single value of approximated pairwise Cramér distance
+#' between q_F and q_G
+#' @details This is function requires the two vectors of quantiles to be
+#' of equal length. The approximation methods are formulated based on
+#' two sets of quantiles corresponding to equally-spaced probability levels.
+#' The approximation formula for "first_rule" is
+#' \deqn{ \text{CD}(F,G) \approx \left\{\frac{1}{K(K+1)}\sum^{2K-1}_{i=1}b_i(b_i+1)(q_{i+1}-q_i)\right }{ (non-Latex version) }
+#' and the approximation formula for "second_rule" is
+#' \deqn{ \text{CD}(F,G) \approx \left\{\frac{1}{(K+1)^2}\sum^{2K-1}_{i=1}b_i^2(q_{i+1}-q_i)\right. }{ (non-Latex version) }
+#' where \eqn{q_i} is an element in a vector of an ordered pooled quantiles
+#' of `q_F` and `q_G` and \eqn{b_i} is an element of a vector of the absolute
+#' values of cumulative sums of \eqn{\mathbf{a}}, whose element is 1 if
+#' \eqn{q_i} is a quantile of F or -1 if \eqn{q_i} is a quantile of G.
+#' @example
+#' f_vector <- 1:10
+#' g_vector <- seq(2,20,2)
+#' pairwise_cd_equal(f_vector,g_vector,"second_rule")
+#' @export
+#'
+pairwise_cd_equal <- function(q_F, q_G, approx_rule) {
```

Comment:
I understand that it doesn't matter what the probability levels are, but maybe we should still ask for them? even if just a single tau argument that would apply to both q_F and q_G?

Reply:
Seems like it is something we should still be thinking about validating.

Reply:
We could, but that information will not be used at all anywhere in the function. Maybe we will have a use for that in an aggregation function (and can ask for it then)?

Reply:
validating = making sure the quantiles from the two forecasts correspond to the same probability levels? If so, then we can add tau_F and tau_G arguments